src/main/java/org/osc/controller/nsc/api/NeutronSdnRedirectionApi.java

```diff
@@ -30,57 +36,78 @@
 import org.osc.sdk.controller.element.NetworkElement;
 import org.osc.sdk.controller.element.VirtualizationConnectorElement;
 import org.osc.sdk.controller.exception.NetworkPortNotFoundException;
+import org.osgi.service.transaction.control.TransactionControl;
 
 public class NeutronSdnRedirectionApi implements SdnRedirectionApi {
 
-    Logger log = Logger.getLogger(NeutronSdnRedirectionApi.class);
-
+	private static final Logger LOGGER = Logger.getLogger(NeutronSdnRedirectionApi.class);
+	
     private VirtualizationConnectorElement vc;
     private String region;
+    
+    private TransactionControl txControl;
+    private EntityManager em;
+    private NSCUtils utils;
 
     public NeutronSdnRedirectionApi() {
     }
 
-    public NeutronSdnRedirectionApi(VirtualizationConnectorElement vc, String region) {
+    public NeutronSdnRedirectionApi(VirtualizationConnectorElement vc, 
+    								String region, 
+    								TransactionControl txControl, 
+    								EntityManager em) {
         this.vc = vc;
         this.region = region;
+        this.txControl = txControl;
+        this.em = em;
+        this.utils = new NSCUtils(em, txControl);
     }
 
     @Override
-    public InspectionHookElement getInspectionHook(NetworkElement inspectedPort,
-            InspectionPortElement inspectionPort) throws NetworkPortNotFoundException, Exception {
-        try (NeutronSecurityControllerApi neutronApi = new NeutronSecurityControllerApi(new Endpoint(this.vc))){
-            return neutronApi.getInspectionHookByPorts(this.region, inspectedPort.getElementId(), inspectionPort);
+    public InspectionHookElement getInspectionHook(NetworkElement inspectedPort, InspectionPortElement inspectionPort)
+            throws Exception {
+//        try (NeutronSecurityControllerApi neutronApi = new NeutronSecurityControllerApi(new Endpoint(this.vc))) {
+//            return neutronApi.getInspectionHookByPorts(this.region, inspectedPort.getElementId(), inspectionPort);
+//        }
+    	CriteriaBuilder cb = this.em.getCriteriaBuilder();
+    	String inspectedPortId = (inspectedPort != null? inspectedPort.getElementId() : null);
+        CriteriaQuery<InspectionHookNSCEntity> q = cb.createQuery(InspectionHookNSCEntity.class);
+        Root<InspectionHookNSCEntity> r = q.from(InspectionHookNSCEntity.class);
+        q.where(cb.equal(r.get("inspectedPortId"), inspectedPortId));
+        			   
+
+        try {
+        	InspectionHookNSCEntity entity = this.em.createQuery(q).getSingleResult();
+        	return NSCUtils.makeInspectionHookElement(entity);
```

Comment:
instead of having the existing InspectionHook and inspectionPort classes inside nsc.model package just delete that package entirely with those classes and make the new db entities simply implement the *Element interfaces. With the new db entities now we should no longer need the existing model classes, the entities can server dual purpose: db persistence and implement the corresponding *Element interfaces.